site/app/libraries/Access.php

```diff
@@ -914,8 +914,17 @@ public function resolveDirPath(string $dir, string $path) {
         }
         $path = implode(DIRECTORY_SEPARATOR, $parts);
 
-        //Make sure it starts with the dir base
-        if (!str_starts_with($path, $info["base"])) {
+        // Diffferent base path for downloading output.csv of rainbow grades
+        if ($dir === "rainbow_grades" && substr($path, -11) === "/output.csv") {
+            // Removing "/upload" from path
+            $base = substr($info["base"],0,-23) . "/rainbow_grades";
+            //checking the path
+            if(!str_starts_with($path, $base))
+            {
+                $path = $base . $path;
+            }
```

Comment:
Is this change just to remove the `/upload` part from the path? You should be able to just add `/upload` in the JS call.

Reply:
yes, it is to remove the `/upload` part from path. Otherwise I had to change the base path of `rainbow_grades` directory
``` php
private function loadDirectories() {
...
        $this->directories["rainbow_grades"] = [
            "base" => $this->core->getConfig()->getCoursePath() . "/uploads/rainbow_grades",
            "subparts" => [],
            "permissions" => [
                "path.read" => "path.read.rainbow_grades",
                "path.write" => "path.write.rainbow_grades",
            ]
        ];
...
}
```
to bypass the checks of `Access.php`. This will effect other URLs as well that's why I thought this will be a faster solution. If you insist I can make those changes by changing the base path.
> @TusharPuri10
> 
> In the original issue I requested this be placed on the "Gradebook" page, not on the "Grade Reports" page. The gradebook page link appears on the left sidebar when "Display Rainbow Grades" is selected on "Course settings"
> 
> Thanks.

 I will also address this issue.